src/flask_cognito_lib/services/cognito_svc.py

```diff
@@ -45,21 +47,32 @@ def get_sign_in_url(
         str
             A front channel login URL for the AWS Cognito AUTHORIZE endpoint
         """
-        quoted_redirect_url = quote(self.cfg.redirect_url)
-
-        full_url = (
-            f"{self.cfg.authorize_endpoint}"
-            f"?response_type=code"
-            f"&client_id={self.cfg.user_pool_client_id}"
-            f"&redirect_uri={quoted_redirect_url}"
-            f"&state={state}"
-            f"&nonce={nonce}"
-            f"&code_challenge={code_challenge}"
-            "&code_challenge_method=S256"
-        )
+
+        query = {
+            "response_type": "code",
+            "client_id": self.cfg.user_pool_client_id,
+            "redirect_uri": self.cfg.redirect_url,
+            "state": state,
+            "nonce": nonce,
+            "code_challenge": code_challenge,
+            "code_challenge_method": "S256",
+        }
 
         if scopes is not None:
-            full_url += f"&scope={'+'.join(scopes)}"
+            query.update({"scope": " ".join(scopes)})
+        if extra_authorize_params is not None:
+            query.update(
+                {
+                    key: value
+                    for key, value in asdict(extra_authorize_params).items()
+                    if value is not None
+                }
+            )
+
+        query_string = urlencode(query, safe="/")
+        full_url = urljoin(
+            self.cfg.authorize_endpoint, "?" + query_string, allow_fragments=False
```

Comment:
```suggestion
            self.cfg.authorize_endpoint, "?" + query_string, allow_fragments=False,
```

Could you add a trailing comma here please, then run `make format` - it means black will then format this out to have one argument per line